p/api/fever.php

```diff
@@ -69,11 +69,11 @@ public function findEntries(array $feed_ids, array $entry_ids, $max_id, $since_i
 		if (!empty($entry_ids)) {
 			$bindEntryIds = $this->bindParamArray('id', $entry_ids, $values);
 			$sql .= " id IN($bindEntryIds)";
-		} else if (!empty($max_id)) {
+		} elseif ($max_id !== null) {
 			$sql .= ' id < :id';
 			$values[':id'] = $max_id;
 			$order = ' ORDER BY id DESC';
-		} else {
+		} elseif ($since_id !== null) {
```

Comment:
This was wrong before, so just the question if we want to make it fail safe:
Looks like the calling part makes sure that either since_id or max_id is not null. But if we consider that someone is trying to use the API programmatically, then we should move both the WHERE and the ORDER (or at least the WHERE) clause into the if-else. Otherwise the SQL would be broken if someone calls this function with max_id=null and since_id=null. 
Or probably use an array to collect all WHERE parts and append it with `implode(' AND ', $where)` before $order is added to $sql.

Reply:
Good point. Here is a suggestion https://github.com/FreshRSS/FreshRSS/pull/1964/commits/44d3e5172655d2d9be4ae5ac383371b251eeae62

Reply:
The old 1=1 trick is also fine :-D